src/main/java/juja/sqlcmd/DatabaseManager.java

```diff
@@ -35,13 +36,52 @@ public boolean connect(String database, String user, String password) {
         }
     }
 
+    public DataSet[] getTableData(String tableName) throws SQLException {
+        if (!hasTable(tableName)) return new DataSet[0];
+        int tableSize = numberOfEntries(tableName);
+        if (tableSize == 0) return new DataSet[0];
+        String sqlQuery = String.format("SELECT * FROM %s", tableName);
+        try (Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery(sqlQuery)) {
+            ResultSetMetaData rsmd = resultSet.getMetaData();
+            int rowSize = rsmd.getColumnCount();
+            return tableRows(resultSet, rowSize, tableSize);
+        }
+    }
+
+    public void close() throws SQLException {
+        if (connection != null) connection.close();
```

Comment:
The method close() must close the current connection, if it is open.
How do we verify that it is open?
Connection is not equals null after invoke close method and we can close an already closed connection

Reply:
Well if the connection is closed then connection.close() does nothing and it's ok. This behavior is acc. to method close() specification:
"Calling the method close on a Connection object that is already closed is a no-op."
So, I should just care about null state of the connection to not have a NPE.